### PR TITLE
Fix JPA storage test issues, small refactoring

### PR DESCRIPTION
**Pull request by jsenko**

### CODE REVIEW

File: storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/JPARegistryStorage.java [hunk 1]
```diff
@@ -595,8 +610,8 @@ public void deleteArtifactVersionMetaData(String artifactId, long version) throw
     public List<RuleType> getGlobalRules() throws RegistryStorageException {
         try {
             return entityManager.createQuery("SELECT r.name FROM Rule r " +
-                    "WHERE r.artifactId IS NULL", RuleType.class)
-                    .getResultList();
+                    "WHERE r.artifactId IS NULL", String.class)
+                    .getResultList().stream().map(e -> RuleType.fromValue(e)).collect(Collectors.toList());
```
Comment (alesj) [inline storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/JPARegistryStorage.java]: map function can be  simplified

File: storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/entity/Rule.java [hunk 2]
```diff
@@ -56,5 +56,5 @@
     private String artifactId;
 
     @Column(name = "name", updatable = false, nullable = false)
-    private RuleType name;
+    private String name;
```
Comment (alesj) [inline storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/entity/Rule.java]: Why plain string and not enum?

File: storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/JPARegistryStorage.java [hunk 3]
```diff
@@ -336,9 +357,9 @@ public void updateArtifactMetaData(String artifactId, EditableArtifactMetaDataDt
             _ensureArtifactExists(artifactId);
 
             return entityManager.createQuery("SELECT r.name FROM Rule r " +
-                    "WHERE r.artifactId = :artifact_id", RuleType.class)
+                    "WHERE r.artifactId = :artifact_id", String.class)
                     .setParameter("artifact_id", artifactId)
-                    .getResultList();
+                    .getResultList().stream().map(e -> RuleType.fromValue(e)).collect(Collectors.toList());
```
Comment (alesj) [inline storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/JPARegistryStorage.java]: fn ...

**Comment by alesj** [comment]
@jsenko why the move of exceptions into separate package?

**Comment by jsenko** [comment]
> @jsenko why the move of exceptions into separate package?

It's just to have them together for better visibility and structure, if you think the change is unnecessary I can revert it.

File: storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/entity/Rule.java (hunk 2 above)
Comment (jsenko) [reply storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/entity/Rule.java]: Hibernate had some problems with the enum, causing errors. I can probably look into in and use the enum if you think it's better that way.

File: storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/JPARegistryStorage.java (hunk 1 above)
Comment (jsenko) [reply storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/JPARegistryStorage.java]: done

File: storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/JPARegistryStorage.java (hunk 3 above)
Comment (jsenko) [reply storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/JPARegistryStorage.java]: done

**Pull request merged**